Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -1342,40 +1339,377 @@ namespace OpenInfraPlatform
+				* \param[in] senseAgreement			Does the resulting geometry have the same sense agreement as the \c IfcOffsetCurve3D.
+				*/
+				void convertIfcOffsetCurve3D(
+					const EXPRESSReference<typename IfcEntityTypesT::IfcOffsetCurve3D>& offsetCurve3D,
+					std::vector<carve::geom::vector<3>>& targetVec,
+					std::vector<carve::geom::vector<3>>& segmentStartPoints,
+					const std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcTrimmingSelect>>& trim1Vec,
+					const std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcTrimmingSelect>>& trim2Vec,
+					const bool senseAgreement,
+					const typename IfcEntityTypesT::IfcTrimmingPreference& trimmingPreference
+				) const noexcept(false)
+				{
+					// **************************************************************************************************************************
+					//	https://standards.buildingsmart.org/IFC/RELEASE/IFC4_3/lexical/IfcOffsetCurve3D.htm
+					//	ENTITY IfcOffsetCurve3D
+					//		SUBTYPE OF(IfcOffsetCurve);
+					//			Distance: IfcLengthMeasure;
+					//			SelfIntersect: IfcLogical;
+					//			RefDirection: IfcDirection;
+					//		WHERE
+					//			DimIs2D : BasisCurve.Dim = 3;
+					//	END_ENTITY;
+					// **************************************************************************************************************************
+					EXPRESSReference<typename IfcEntityTypesT::IfcCurve> basisCurve = offsetCurve3D->BasisCurve;
+					std::vector<carve::geom::vector<3>> basisCurvePoints;
+					double distance = offsetCurve3D->Distance * this->UnitConvert()->getLengthInMeterFactor();
+					
+					EXPRESSReference<typename IfcEntityTypesT::IfcDirection> offsetDirectionVector = offsetCurve3D->RefDirection;
+					carve::geom::vector<3> direction = placementConverter->convertIfcDirection(offsetCurve3D->RefDirection);
+
+
+					convertIfcCurve(basisCurve, basisCurvePoints, segmentStartPoints,
+						trim1Vec, trim2Vec, senseAgreement, trimmingPreference);
+
+					std::vector<carve::geom::vector<3>> offsetCurve3DPoints;
+
+					for (int i = 0; i < basisCurvePoints.size() - 1; i++) {
+						carve::geom::vector<3> startPoint = basisCurvePoints[i];
+						carve::geom::vector<3> endPoint = basisCurvePoints[i + 1];
+
+						carve::geom::vector<3> tangent = endPoint - startPoint;
+
+						carve::geom::vector<3> offset = carve::geom::cross(direction, tangent);
+						carve::geom::vector<3> offsetPoint = startPoint + offset * distance;
+						offsetCurve3DPoints.push_back(offsetPoint);
+					}
+					
+					carve::geom::vector<3> lastStartPoint = basisCurvePoints[basisCurvePoints.size() - 1];
+					carve::geom::vector<3> lastEndPoint = basisCurvePoints[basisCurvePoints.size() - 2];
+					carve::geom::vector<3> lastTangent = lastEndPoint - lastStartPoint;
+
+					carve::geom::vector<3> lastOffset = carve::geom::cross(direction, lastTangent);
+					carve::geom::vector<3> lastOffsetPoint = lastStartPoint - lastOffset;

```

Comment:
I think it should be 
```suggestion
					carve::geom::vector<3> lastOffsetPoint = lastStartPoint - lastOffset * distance;

```
```